io/src/main/scala/sbt/io/IO.scala

```diff
@@ -551,16 +552,16 @@ object IO {
   }
 
   /** Deletes `file`, recursively if it is a directory. */
-  def delete(file: File): Unit = retry {
+  def delete(file: File): Unit = Retry {
     try {
-      FileTreeView.DEFAULT.list(Glob(file, AllPassFilter, 0)).foreach {
-        case dir if dir.isDirectory => delete(dir.toPath.toFile)
-        case f                      => f.toPath.toFile.delete()
+      FileTreeView.default.list(file.toPath).foreach {
+        case (dir, attrs) if attrs.isDirectory => delete(dir.toFile)
+        case (f, _)                            => Files.deleteIfExists(f)
       }
     } catch {
       case _: NotDirectoryException =>
     }
-    file.delete()
+    Files.deleteIfExists(file.toPath)
```

Comment:
Maybe Windows is not ready for `Files.deleteIfExists(file.toPath)`?

Reply:
No. I reverted that and it made no difference. Based on the error message, underlying problem is that ScriptedTests is trying to delete a file that is still in use by the forked scripted process. The question is why? I also believe that somehow scripted is leaking classloaders because I've noticed a marked increase in the frequency off my running out of metaspace when running scripted in a continuous build.

Reply:
Looks like it was part of https://github.com/sbt/sbt/pull/3151, which allows scripted to reuse sbt instance.

Reply:
https://github.com/sbt/sbt/blob/bc21db28646f0b4a4484cdd22b6a4c4596cafde1/scripted-sbt-redux/src/main/scala/sbt/scriptedtest/ScriptedTests.scala#L329-L332

```scala
          // Run the test and delete files (except global that holds local scala jars)
          val result = runOrHandleDisabled(label, tempTestDir, runTest, buffer)
          IO.delete(tempTestDir.*("*" -- "global").get)
          result
```

The code has existed for a while, so what's newly holding on to the `project\target\config-classes\$17a69f0703746b7bed64$.class`?